### DISCUSSION TITLE
How do mentions and quotes work?

### DISCUSSION THREAD

**Discussion by jeffxt** [bbernhard/signal-cli-rest-api#334] [Q&A] [answered]
I checked the [Swagger](https://bbernhard.github.io/signal-cli-rest-api/#/Messages/post_v2_send) but I can't find any documentation on how mentions or quotes work in the `/v2/send` endpoint.  

For context, I found another discussion post showing how mentions work, but this references the `signal-cli`  https://github.com/bbernhard/signal-cli-rest-api/discussions/302. 

Does anyone have any more information on this?

**Comment by RoadRunner-** [comment]
The `v1/about` enpoints shows mentions as capability, but im not sure if this capabilities are from the underlying signal-cli?
```
{
    "versions": [
        "v1",
        "v2"
    ],
    "build": 2,
    "mode": "normal",
    "version": "0.65",
    "capabilities": {
        "v2/send": [
            "quotes",
            "mentions"
        ]
    }
}
```

I also couldn`t find any documentation about how to use this feature via REST API. There is only the [description](https://github.com/AsamK/signal-cli/blob/master/man/signal-cli.1.adoc#send) from signal-cli.

**Comment by RoadRunner-** [comment, marked as answer]
I just had a look at the code. The structures already support parameters for mentions (https://github.com/bbernhard/signal-cli-rest-api/blob/master/src/api/api.go#L100 and https://github.com/bbernhard/signal-cli-rest-api/blob/master/src/client/client.go#L62)

Using Postman, I was able to successfully send a POST message with the following body. 
```
{
    "number": "<sender>",
    "recipients": 
    [
        "<group-id>"
    ],
    "message": "Hi X!",
    "mentions": [
        {
            "start": 3,
            "length": 1,
            "author": "<receiver>"
        }
    ]
}
```

I guess it just needs an update for the documentation/swagger.

**Reply by jeffxt** [reply]
> I just had a look at the code. The structures already support parameters for mentions (https://github.com/bbernhard/signal-cli-rest-api/blob/master/src/api/api.go#L100 and https://github.com/bbernhard/signal-cli-rest-api/blob/master/src/client/client.go#L62)
> 
> Using Postman, I was able to successfully send a POST message with the following body.
> 
> ```
> {
>     "number": "<sender>",
>     "recipients": 
>     [
>         "<group-id>"
>     ],
>     "message": "Hi X!",
>     "mentions": [
>         {
>             "start": 3,
>             "length": 1,
>             "author": "<receiver>"
>         }
>     ]
> }
> ```
> 
> I guess it just needs an update for the documentation/swagger.

This worked perfectly, thank you!  I appreciate you taking the time to post a solution!

Tip for future readers:  for a "clean-looking" message, you can start a message using an @ mention, like you might do when typing it out a message.  Just set the `start` position to 0 and start your message with a placeholder like X, as in the example above. 

For example:

> X thanks for your help today!


becomes

> @RoadRunner thanks for your help today!
